### PR TITLE
Consistent type print

### LINKED ISSUE #9501: Generic is formatted differently between babel and typescript
**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEUCGBbOJTAZQgIHEE4AnASzAB4AdKTTABXoGsGBhaAGa0A5gBpMTEAHkAcgFEA+gBlZ5BewBKU9pMwAfCSACCMqQBUAEnI3LV6rTpAA+Fk8wBebAFcANj4DcLCCiIBAADjC06MigONTUEADu7HEIaMggOD6JOACe6SEARtQ4YFxwMMRhpbRQwsgw1F5wIXB4hXAAJp1dSjh1XjjCcABiENR4ODCRdRk4XjAQwSAAFjB4PgDqK7TwaNVgcMRpu7QAbru5GWBoBSC1aDQw7CXCk8gCWY8hAFZoAB4AIRKZQqxHwcCUtTgHy+LRAf3+xFqwh8cAAil4IPBYT5viBqtRHtQMoUcB0fMswnRYJtaJ0YCtkAAOAAMIWpEEemxKYQy1LgxLOMJCAEcsfAXuF0igcGgALRQOBdLrLahwcW0dUvIbvJCfPHwx54WgNJpGlFozHYmH6uEhGDkukMplIABMDpKtB8KP4eD1IEFAFZll5HmZyTKDfizs0AJJQHqwYhgOgRIyJ4gwXJo3GPAC++aAA)
```sh
--parser babel
```

**Input:**
```jsx
const name: SomeGeneric<
  Pick<Config, "ONE_LONG_PROP" | "ANOTHER_LONG_PROP">
> = null;

```

**Output:**
```jsx
const name: SomeGeneric<
  Pick<Config, "ONE_LONG_PROP" | "ANOTHER_LONG_PROP">
> = null;

```

While using `typescript` parser:

[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEUCGBbOJTAZQgIHEE4AnASzAB4AdKTTABXoGsGBhaAGa0A5gBpMTEAHkAcgFEA+gBlZ5BewBKU9pMwAfCSACCMqQBUAEnI3LV6rTpAA+Fk8wBebAFcANj4DcLCCiIBAADjC06MigONTUEADu7HEIaMggOD6JOACe6SEARtQ4YFxwMMRhpbRQwsgw1F5wIXB4hXAAJp1dSjh1XjjCcABiENR4ODCRdRk4XjAQwSAAFjB4PgDqK7TwaNVgcMRpu7QAbru5GWBoBSC1aDQw7CXCk8gCWY8hAFZoAB4AIRKZQqxHwcCUtTgHy+LRAf3+xFqwh8cAAil4IPBYT5viBqtRHtQMjBcmE4GgwHQIsswnRYJtaJ0YCtkAAOAAMIXpEEemxKYQy9MpNDOMJCAEcsfAXuF0igcGgALRQOBdLrLahwaW0bUvIbvJCfPHwx54WgNJpmlFozHYmHGuEhGA4QpMllspAAJhdJVoPhR-DwRpAlIArMsvI8zG6FSb8WdmgBJKA9WDEam0CJGNPEMlo3GPAC+xaAA)
```sh
--parser typescript
```

**Output:**
```tsx
const name: SomeGeneric<Pick<
  Config,
  "ONE_LONG_PROP" | "ANOTHER_LONG_PROP"
>> = null;

```

**Expected behavior:**

I'd expect the 2 parsers to result in the same output. Which one of them I don't really have a preference, but I guess Babel's behavior as it's fewer lines of code and it's easier to read the `Pick` on its own line.

(FWIW, `babel-flow` and `flow` is consistent with `babel` and `babel-ts` is consistent with `typescript`).

### PR DESCRIPTION
**Pull request by fisker**
This PR is more like #6467 and #6901 rewrite, previous attempt #9520

Fixes #9501

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/typescript/decorators/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -389,9 +389,8 @@ class Bar {
 }
 
 class MyContainerComponent {
-  @ContentChildren(MyComponent) components: QueryListSomeBigName<
-    MyComponentThat
-  >;
+  @ContentChildren(MyComponent)
+  components: QueryListSomeBigName<MyComponentThat>;
 }
```
Comment (fisker) [inline tests/typescript/decorators/__snapshots__/jsfmt.spec.js.snap:394]: **Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgLICeAwtDOgJZRwBOJAtgA7QIw7AA6UOOkjzsmJDgCKAVxoEAMuUwwAyhDpwAQuQDmAOXRKxE6bIVLVm7XF3UpM+YpXqtSgDydu3QvSZVYAFQAW6GM44AHwA3OwBUAC+IAA0IBAMMOTQmMig6NTUEADuAAoZCKko6KjZ6ASpcQBG1OhgANZw8gx1lGrIMNTicXB0VXAAJgODkuhQaqLoanAAYhDUdP5J48gg6KIwELEgPjB0qADqPuTwmC1gcHKFJ+QAbicEq2DY25SYNDC5tWqLyABmJXecQAVpgAB7KWoNJpyUzSKj-QFwEHguRtVBmUQQeCI1BAkAtajvairP6oHLbBjUSgwA7kAYwHzIAAcAAY4lSIO8DrUGKsqXBibc4NsAI5Y+BfBJFNaYAC0VEGg221Dg4vIqq+U1+SABeORIHedHIHS6Bsw6Mx2JFuqRcTIVTpDKZSAATPbauRUG16DqQIKAKzbUTvLzoKpFPX427iACSUGGsDkYGpiQAggm5DACBjce9IpEgA)
```sh
--parser flow
```

**Input:**
```jsx
class MyContainerComponent {
  components: QueryListSomeBigNameQueryListSomeBigNameQueryListSomeBigName<
    MyComponentThat
  >;	
}
```

**Output:**
```jsx
class MyContainerComponent {
  components: QueryListSomeBigNameQueryListSomeBigNameQueryListSomeBigName<MyComponentThat>;
}

```

**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgLICeAwtDOgJZRwBOJAtgA7QIw7AA6UOOkjzsmJDgCKAVxoEAMuUwwAyhDpwAQuQDmAOXRKxE6bIVLVm7XF3UpM+YpXqtSgDydu3QvSZVYAFQAW6GM44AHwA3OwBUAC+IAA0IBAMMOTQmMig6NTUEADuAAoZCKko6KjZ6ASpcQBG1OhgANZw8gx1lGrIMNTicXB0VXAAJgODkuhQaqLoanAAYhDUdP5J48gg6KIwELEgPjB0qADqPuTwmC1gcHKFJ+QAbicEq2DY25SYNDC5tWqLyABmJXecQAVpgAB7KWoNJpyUzSKj-QFwEHguRtVBmUQQeCI1BAkAtajvairGAEBhwTBgajkRLbBg02AHcgDGA+ZAADgADHEGRB3gdagxVgzKTRbnBtgBHLHwL4JIprTAAWiog0G22ocBl5C1XymvyQALxyJA7zo5A6XVNmHRmOxkqNSLiZCqzNZ7KQACYXbVyKg2vRDSBKQBWbaid5edBVIrG-G3cQASSgw1gcmptJgAEFU3IyRjce9IpEgA)
```sh
--parser typescript
```

**Input:**
```tsx
class MyContainerComponent {
  components: QueryListSomeBigNameQueryListSomeBigNameQueryListSomeBigName<
    MyComponentThat
  >;	
}
```

**Output:**
```tsx
class MyContainerComponent {
  components: QueryListSomeBigNameQueryListSomeBigNameQueryListSomeBigName<
    MyComponentThat
  >;
}

```

File: tests/typescript/comments/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -702,9 +702,7 @@ interface Foo {
 }
 type T = </* comment */>(arg) => any;
 
-functionName<
-  A // comment
->();
+functionName<A>(); // comment
```
Comment (fisker) [inline tests/typescript/comments/__snapshots__/jsfmt.spec.js.snap:705]: **Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0ByAhgLZwA8AOlAATUCC1A9I9ZMabBTFQHwAUASgDcIADQgIAB1zQAzslCEATkogB3AArKE8lIQA2awgE954gEZLCYANZwYAZUnWcUAObIYS9HHFxi5nAAJkHBADKE7uiEbnAAYhBKxIQwuO7IIIToMBBiIAAWMMT6AOr5OPCyzmBwDjoVOABuFcYZYLJmIK6ycEowGlZuycioBj3iAFayAB4AQla29g4kcGGucCNjviBT0w6ubvpwAIroEPCb+uMgzko9Shmo+up5kkquMCU4QTD5yAAcAAZxG8ID0SlZJBk3nB7o0NuIAI5neADKS6TKyAC0UDgwWCeSUcGROCJAxiwyQoyu2x6xBwnm8tIOR1O5w2VK24hghHMXx+fyQACZuVYcPoDgBhCDsQgZWEAVjy6B6ABVebpqddGj4AJJQUKwBxgd7SOgGhwwYxHS49AC+dqAA)
```sh
--parser flow
```

**Input:**
```jsx
functionName<
  A // comment	
>();
```

**Output:**
```jsx
functionName<A>(); // comment

```

**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0ByAhgLZwA8AOlAATUCC1A9I9ZMabBTFQHwAUASgDcIADQgIAB1zQAzslCEATkogB3AArKE8lIQA2awgE954gEZLCYANZwYAZUnWcUAObIYS9HHFxi5nAAJkHBADKE7uiEbnAAYhBKxIQwuO7IIIToMBBiIAAWMMT6AOr5OPCyzmBwDjoVOABuFcYZYLJmIK6ycEowGlZuycioBj3iAFayAB4AQla29g4kcGGucCNjviBT0w6ubvpwAIroEPCb+uMgzko9ShkwxpJwsmBKONJ5kh+wJThBGD5ZAADgADOIfhAeiUrJIMj9Xr1GhtxABHM7wAZSXSZWQAWigcGCwTySjgGJw5IGMWGSFGV22PWIOE83iZByOp3OG3pW3EMEI5n+gOBSAATAKrDh9AcAMIQdiEDKvACseXQPQAKkLdAzro0fABJKChWAOd6fGB0U0OJ5HS49AC+TqAA)
```sh
--parser typescript
```

**Input:**
```tsx
functionName<
  A // comment	
>();
```

**Output:**
```tsx
functionName<
  A // comment
>();

```

File: tests/typescript/generic/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -73,9 +73,7 @@ export const getVehicleDescriptor = async (
 
 export const getVehicleDescriptor = async (
   vehicleId: string
-): Promise<
-  Collections.Parts.PrintedCircuitBoardAssemblyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy
-> => {};
+): Promise<Collections.Parts.PrintedCircuitBoardAssemblyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy> => {};
```
Comment (fisker) [inline tests/typescript/generic/__snapshots__/jsfmt.spec.js.snap:76]: **Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACSKAZ3wHM4YA1OACwEswAbOAETiLGzoxhzwF48AQyIBPKGDwAKADpQ8eAG60GzAJIATJHhJcopGTDkBKbQAVsEALZ0icADxyFAYQiNmYGHWhEAdGaFcPws6WDgNZzpsMABXOhgAIQhAjQBBIjsrACNGUTz8gsKi4qLDOQA+AUrgAF8AbhAAGhAIHm9iZFBAywB3AOwEImQQIUYeoVEh5qzsITAAawoAZQw50NJkGGwYuGa4bPCNcIAZIX0YoXIAMRwrIRgvfWGhGN4mkBoYK0YAdXp4IirMBwJaDeJ0RTxUTDMAZd6hOy4CyXO7IABmozszQAVkQ0AlZgtlkIrHBjqE4OjMbsQLi0Et1swAIoxCDwKmMLEgVbYRHDNGMCA9d4YPQwH50DQwGjIAAcAAZmqKIHYfrMMMNRew4NhlO8AI6s+AWVpDFAiAC0UDg4XC7wGhqicGRpFRSAxnJpmTom22XsZcBZbMp7upzRgQiyEqlMqQACZw7M6Ix1q4rG7UEQAKzvGJ2AAqkbNHq5ih2aigR1gS043BgqUrSxgomYHLsNRqQA)
```sh
--parser flow
```

**Input:**
```jsx
export const getVehicleDescriptor = async (
  vehicleId: string	
): Promise<
  Collections.Parts.PrintedCircuitBoardAssemblyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy	
> => {};
```

**Output:**
```jsx
export const getVehicleDescriptor = async (
  vehicleId: string
): Promise<Collections.Parts.PrintedCircuitBoardAssemblyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy> => {};

```

**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACSKAZ3wHM4YA1OACwEswAbOAETiLGzoxhzwF48AQyIBPKGDwAKADpQ8eAG60GzAJIATJHhJcopGTDkBKbQAVsEALZ0icADxyFAYQiNmYGHWhEAdGaFcPws6WDgNZzpsMABXOhgAIQhAjQBBIjsrACNGUTz8gsKi4qLDOQA+AUrgAF8AbhAAGhAIHm9iZFBAywB3AOwEImQQIUYeoVEh5qzsITAAawoAZQw50NJkGGwYuGa4bPCNcIAZIX0YoXIAMRwrIRgvfWGhGN4mkBoYK0YAdXp4IirMBwJaDeJ0RTxUTDMAZd6hOy4CyXO7IABmozszQAVkQ0AlZgtlkIrHBjqE4OjMbsQLi0Et1swAIoxCDwKmMLEgVbYRHDGCiDDsTjcGDvDB6GA-OgaGA0ZAADgADM0JRA7D9ZhhhhL2HBsMp3gBHVnwCytIYoEQAWigcHC4XeAxNUTgyNIqKQGM5NMydE2219jLgLLZlK91OaMCEWWlsvlSAATFHZnRGOtXFZPagiABWd4xOwAFRjlu9XMUOzUUCOsCWIp4qRrSwFzA5dhqNSAA)
```sh
--parser typescript
```

**Input:**
```tsx
export const getVehicleDescriptor = async (
  vehicleId: string	
): Promise<
  Collections.Parts.PrintedCircuitBoardAssemblyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy	
> => {};
```

**Output:**
```tsx
export const getVehicleDescriptor = async (
  vehicleId: string
): Promise<
  Collections.Parts.PrintedCircuitBoardAssemblyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy
> => {};

```

File: tests/typescript/template-literals/__snapshots__/jsfmt.spec.js.snap [hunk 4]
```diff
@@ -39,9 +39,7 @@ printWidth: 80
 =====================================input======================================
 const bar = tag<number>\`but where will prettier wrap such a long tagged template literal? \${foo.bar.baz} long long long long long long long long long long long long long long\`;
 =====================================output=====================================
-const bar = tag<
-  number
->\`but where will prettier wrap such a long tagged template literal? \${foo.bar.baz} long long long long long long long long long long long long long long\`;
+const bar = tag<number>\`but where will prettier wrap such a long tagged template literal? \${foo.bar.baz} long long long long long long long long long long long long long long\`;
```
Comment (fisker) [inline tests/typescript/template-literals/__snapshots__/jsfmt.spec.js.snap:42]: **Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAjAhgJ0wF5MZcBzAHgB0pNMoBXAW2zn2ploD4ADbRlgDuAC3ZxMQgJYAbGZgAO+ODBhT2k-LgWY0jMCMy5MM6OVIVycACak4zBTNzwTU+FpkB+TNWoASYAAzCAgAOjx8cNwALwBfEzMEqHNTZKSUxNSMtKz0vNyCzLNeAG4QABoQCAU1dGRQAnwIIQAFAgQ0ZBBcGSFcAE9OyuwtMABrFQBlBVwwKWTkGHxGOEr7NmtrGwAZXGTGCjgAMQh8Zmc1BZRcQQgKkBEYZhkAdRE3ODQZsDhJjrcpAA3Nz9LpgNBDEDzNDsGAtLTkc7IQI9GGVABWaAAHgAhUYTGCTXDMODbeZwZGo1YgTFYybzcgyOAARUYEHglJkaJAM3wMPwXUCpiE9yU8xgLyk1hgImQAA4AAyVJQQGEvLQKLpKT7sQEUyoARzZ8Hh1U61zQAFooHAbDZ7sojVJlPCKEikCiudSYcwpItlt6GUzWeyKR6qZUyNhJdLZUgAEyRrSyBkAYQgzHdIE+AFZ7owYQAVXDYc2e7mAlYASSgW1gkzA+CkNQAgrXJjB+kzOTDYrEgA)
```sh
--parser flow
```

**Input:**
```jsx
const bar = tag<
  number	
>`but where will prettier wrap such a long tagged template literal? \${foo.bar.baz} long long long long long long long long long long long long long long`;
```

**Output:**
```jsx
const bar =
  tag <
  number >
  `but where will prettier wrap such a long tagged template literal? \${foo.bar.baz} long long long long long long long long long long long long long long`;

```

**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAjAhgJ0wF5MZcBzAHgB0pNMoBXAW2zn2ploD4ADbRlgDuAC3ZxMQgJYAbGZgAO+ODBhT2k-LgWY0jMCMy5MM6OVIVycACak4zBTNzwTU+FpkB+TNWoASYAAzCAgAOjx8cNwALwBfEzMEqHNTZKSUxNSMtKz0vNyCzLNeAG4QABoQCAU1dGRQAnwIIQAFAgQ0ZBBcGSFcAE9OyuwtMABrFQBlBVwwKWTkGHxGOEr7NmtrGwAZXGTGCjgAMQh8Zmc1BZRcQQgKkBEYZhkAdRE3ODQZsDhJjrcpAA3Nz9LpgNBDEDzNDsGAtLTkc7IQI9GGVABWaAAHgAhUYTGCTXDMODbeZwZGo1YgTFYybzcgyOAARUYEHglJkaJAM3wMPwXRg-QUnzA+CkNXuSnmMBeUmsMBEyAAHAAGSpKCAwl5aBRdJSfdiAimVACObPg8OqnWuaAAtFA4DYbPdlOapMp4RQkUgUVzqTDmFJFssAwymaz2RTfVTKmRsHKFUqkAAmONaWQMgDCEGYPpAnwArPdGDCACq4bA2v3cwErACSUC2sEmYolMAAgk3JkKmZyYbFYkA)
```sh
--parser typescript
```

**Input:**
```tsx
const bar = tag<
  number	
>`but where will prettier wrap such a long tagged template literal? \${foo.bar.baz} long long long long long long long long long long long long long long`;
```

**Output:**
```tsx
const bar = tag<
  number
>`but where will prettier wrap such a long tagged template literal? \${foo.bar.baz} long long long long long long long long long long long long long long`;

```

File: src/language-js/utils.js [hunk 5]
```diff
@@ -443,16 +443,28 @@ const flowTypeAnnotations = new Set([
...
 ]);
+const simpleTypeScriptTypeAnnotations = new Set([
+  "TSAnyKeyword",
+  "TSNullKeyword",
+  "TSTypeReference",
+  "TSThisType",
+  "TSNumberKeyword",
+  "TSVoidKeyword",
+  "TSBooleanKeyword",
+  "TSLiteralType",
```
Comment (sosukesuzuki) [inline src/language-js/utils.js:454]: I think we should add `"TSSymbolKeyword"`.
**Prettier pr-9521**
[Playground link](https://deploy-preview-9521--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuc0DOMAEAzCEmYBieJpZ5FlV1NpAPGgJ4C2ARhADYB8A3ADpQQAGhAQADjACW6ZKACGAJwUQA7gAVFCNMhByOKuY20jWCuWADWcGAGUx5yVADmyGAoCucEXDZwAJn7+ADJyzu5yTnDECsxyMFLOOnLuMBDCIAAWMMwcAOoZkvBo9mBwNlqFkgBuhYw6YGjGII5ocAowamZOscjYeq0iAFZoAB4AQmaW1jZyzHBBjnC9-V4gwyM2jk4ccACK7hDwyxwDIPYKrQo6MIxicGhgCpIS6WJPsLmSfjAZyAAcAAYRG8IK1cmYxDo3vc2lUliIAI4HeCdcTaFByNAAWigcH8-nSCjgSMkRM6ER6SD6J1WrWYklcHlpWx2+0OSypKxEMDkrE+31+SAATNyzJIOFsAMIQZiUkD3ACs6XcrQAKrz0dTTlVPABJKCBWA2R7PGAAQQNNhuO2OrQAvnagA)
```sh
--parser typescript
```

**Input:**
```tsx
const foo: Fooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo<symbol>;

```

**Output:**
```tsx
const foo: Fooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo<
  symbol
>;

```

Comment (sosukesuzuki) [inline src/language-js/utils.js:454]: Please add more tests for `TSAnyKeyword`, `TSNullKeyword`, `TSThisType`, `TSVoidKeyword` and `TSLiteralType`.
I ran tests with the below code, but tests successful.
```ts
const simpleTypeScriptTypeAnnotations = new Set([
  // "TSAnyKeyword",
  // "TSNullKeyword",
  "TSTypeReference",
  // "TSThisType",
  "TSNumberKeyword",
  // "TSVoidKeyword",
  "TSBooleanKeyword",
  // "TSLiteralType",
]);
```

File: tests/typescript/comments/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (sosukesuzuki) [reply tests/typescript/comments/__snapshots__/jsfmt.spec.js.snap:705]: I think the current TypeScript format is better. What do you think?

**Review by sosukesuzuki — COMMENTED** [review]
Overall looks good. Please see comments.

File: tests/typescript/comments/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (fisker) [reply tests/typescript/comments/__snapshots__/jsfmt.spec.js.snap:705]: Maybe, but I mean to consistent them with `flow`, let's improve it in another PR?

File: src/language-js/utils.js (hunk 5 above)
Comment (fisker) [reply src/language-js/utils.js:454]: I made this list according to `simpleFlowTypeAnnotations`.

Tests added https://github.com/prettier/prettier/pull/9521/commits/a51fb7f7eb417b55076ca66a2f451a338be3c41a

Comment (fisker) [reply src/language-js/utils.js:454]: Flow format it like this

**Prettier pr-9521**
[Playground link](https://deploy-preview-9521--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuc0DOMAEAzCEmYBieJpZ5FlV1NpAPADpSaZoCeAtgEYQA2TAPkwBeTAEMA3CAA0ICAAcYAS3TJQYgE4aIAdwAKmhGmQgxvHWLbHZXDWLABrODADK8+0qgBzZDA0BXOFk4bjgAEzDwgBkxb38xLzhiDQ4xGGVvEzF-GAgZEAALGA5eAHUCpXg0dzA4FyNKpQA3SrYTMDRrEE80OA0YPTsvVORsM17ZACs0AA8AITtHZxcxDjgozzhR8aCQaZmXTy9eOABFfwh4bd4JkHcNXo0TbF5dfPkNTxhSpTCYAuQAA4AAyyD4QXqlOzyEwfOCPJpbWQARwu8EGCmMKDEaAAtFA4OFwvkNHBUUpSYMEiMkGMbrtehwlL4AgyjidzpctrSdrIYGIuD8-gCkAAmPl2JS8I4AYQgHBpIHhAFZ8v5egAVAVYum3JqBACSUEisBcYE+igAgsaXDA2Cdrr0AL5OoA)
```sh
--parser flow
```

**Input:**
```jsx
const foo: Fooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo<
  symbol
> = a;
```

**Output:**
```jsx
const foo: Fooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo<
  symbol
> = a;

```

I know this is just two types into the list, but I still prefer in another PR.
I added a test for it https://github.com/prettier/prettier/pull/9521/commits/a21ebc28d968c858c41861f54d1fef56ef75f0ff

**Review by sosukesuzuki — APPROVED** [review]
Good job

**Review by evilebottnawi — APPROVED** [review]

File: tests/typescript/comments/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (fisker) [reply tests/typescript/comments/__snapshots__/jsfmt.spec.js.snap:705]: Tracked #9537

**Pull request merged**